app/javascript/src/components/form/Derivatives.svelte

```diff
@@ -0,0 +1,84 @@
+<script>
+  import Tags from './Tags.svelte';
+  import { onMount } from 'svelte';
+
+  let showDerivative = false;
+  let maxCodes = 5;
+  export let currSources = [];
+
+  onMount(() => {
+    if (currSources.length) showDerivative = true;
+  });
+
+  async function handleAutoCompleteRequest(value) {
+    if (!value) return [];
+
+    const timeout = 8000;
+    const timeoutController = new AbortController();
+    const timeoutID = setTimeout(() => timeoutController.abort(), timeout);
+
+    const response = await fetch(`/code/${value}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json'
+      },
+      signal: timeoutController.signal
+    });
+    const text = await response.text();
+    clearTimeout(timeoutID);
+
+    if (response.ok) {
+      const json = JSON.parse(text);
+      return json.map(post => postToResult(post));
+    } else {
+      throw new Error(`${response.status} ${response.statusText}`);
```

Comment:
We might need to actually handle the error. Just setting a variable and displaying it is probably enough

Reply:
The error is caught in the Tags form via the `await` block.

See https://github.com/EloHellEsports/workshop.codes/pull/53/files#diff-cbf984e6496b634b0209250da97c2279b04e079aa8ec2330e0d0e307733c9b4dR205-R213
![image](https://user-images.githubusercontent.com/10406383/125005212-187ee500-e010-11eb-89b8-409c4ef069eb.png)

Reply:
Ah! Perfect